File: app/src/main/java/com/lxk/animandcustomviewdemo/animator/evaluator/CharEvaluator.java
package com.lxk.animandcustomviewdemo.animator.evaluator;

import android.animation.TypeEvaluator;

/**
 * @author https://github.com/103style
 * @date 2019/12/11 15:58
 * <p>
 * 字符变化
 */
public class CharEvaluator implements TypeEvaluator<Character> {

    @Override
    public Character evaluate(float fraction, Character startValue, Character endValue) {
        int s = startValue;
        int e = endValue;
        int cur = s + (int) (fraction * (e - s));
        return (char) cur;
    }
}

File: app/src/main/java/com/lxk/animandcustomviewdemo/animator/evaluator/CharEvaluator.kt
package com.lxk.animandcustomviewdemo.animator.evaluator

import android.animation.TypeEvaluator

/**
 * @author https://github.com/103style
 * @date 2020/1/10 14:45
 * <p>
 * 字符变化
 */
class CharEvaluator : TypeEvaluator<Char> {

    override fun evaluate(fraction: Float, startValue: Char, endValue: Char): Char {
        val s = startValue.toInt()
        val e = endValue.toInt()
        val cur = s + (fraction * (e - s)).toInt()
        return cur.toChar()
    }
}
